packages/back-end/src/controllers/features.ts

```diff
@@ -464,6 +470,44 @@ export async function postFeatureDefaultValue(
   });
 }
 
+export async function postFeatureSchema(
+  req: AuthRequest<{ schema: string; enabled: boolean }, { id: string }>,
+  res: Response<{ status: 200 }, EventAuditUserForResponseLocals>
+) {
+  const { org } = getOrgFromReq(req);
+  const { id } = req.params;
+  const { schema, enabled } = req.body;
+  const feature = await getFeature(org.id, id);
+
+  if (!feature) {
+    throw new Error("Could not find feature");
+  }
+
+  req.checkPermissions("manageFeatures", feature.project);
+  req.checkPermissions("createFeatureDrafts", feature.project);
+
+  const updatedFeature = await setJsonSchema(
+    org,
+    res.locals.eventAudit,
+    feature,
+    schema,
+    enabled
+  );
```

Comment:
Since we have more than 3 positional arguments it might be a good idea to use an options object.